contract/contracts/CashRemunerationEIP712.sol

```diff
@@ -0,0 +1,176 @@
+    function initialize(address owner) public initializer {
+        __Ownable_init(owner);
+        __ReentrancyGuard_init();
+        __Pausable_init();
+        __EIP712_init("CashRemuneration", "1");
+    }
+
+    /**
+     * @dev Computes the hash of a given WageClaim.
+     * @param wageClaim The WageClaim struct to hash.
+     * @return The hash of the WageClaim.
+     */
+    function wageClaimHash(WageClaim calldata wageClaim) private pure returns (bytes32) {
+        return keccak256(abi.encode(
+            WAGE_CLAIM_TYPEHASH,
+            wageClaim.employeeAddress,
+            wageClaim.hoursWorked,
+            wageClaim.hourlyRate,
+            wageClaim.date
+        ));
+    }
+
+    /**
+     * @notice Allows an employee to withdraw their wages.
+     * @dev Hourly rate is in ether so it needs to be multiplied
+     * by 10 ** 18
+     * @param wageClaim The details of the wage being claimed.
+     * @param signature The ECDSA signature.
+     *
+     * Requirements:
+     * - The caller must be the employee specified in the wage claim.
+     * - The wage claim must be signed by the contract owner.
+     * - The wage claim must not have already been paid.
+     * - The contract must have sufficient Ether to fulfill the claim.
+     * - The contract must not be paused.
+     *
+     * Emits a {Withdraw} event.
+     */
+    function withdraw(
+        WageClaim calldata wageClaim,
+        bytes calldata signature
+    ) external whenNotPaused nonReentrant {
+        require(msg.sender == wageClaim.employeeAddress, "Withdrawer not approved");
+
+        bytes32 digest = keccak256(abi.encodePacked(
+            "\x19\x01",
+            _domainSeparatorV4(),
+            wageClaimHash(wageClaim)
+        ));
+
+        address signer = digest.recover(signature);//ecrecover(digest, v, r, s);
+
+        if (signer != owner()) {
+            revert UnauthorizedAccess(owner(), signer);
+        }
+
+        bytes32 sigHash = keccak256(signature);
+
+        require(!paidWageClaims[sigHash], "Wage already paid");
+
+        paidWageClaims[sigHash] = true;
+
+        uint256 amountToPay = wageClaim.hoursWorked * (wageClaim.hourlyRate * 10 ** 18);
+
+        payable(wageClaim.employeeAddress).sendValue(amountToPay);
+
+        emit Withdraw(wageClaim.employeeAddress, amountToPay);
+    }
+
+    /**
+     * @notice Pauses the contract, disabling certain functions.
+     * @dev Can only be called by the contract owner.
+     */
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    /**
+     * @notice Unpauses the contract, enabling paused functions.
+     * @dev Can only be called by the contract owner.
+     */
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    /**
+     * @notice Retrieves the contract's Ether balance.
+     * @return The balance of the contract in wei.
+     */
+    function getBalance() external view returns (uint256) {
+        return address(this).balance;
+    }
```

Comment:
i think we dont have to define this function? since it can be called using getBalance in viem

Reply:
Oh okay, didn't know, will update in future thanks